src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/mha_single_token.cpp

```diff
@@ -146,27 +617,67 @@ void mha_single_token_kernel(const ov::intel_cpu::PlainTensor& query,
     }
     if (d_scale == 0.0f)
         d_scale = 1.0f / sqrt(S);
+    auto nthr = parallel_get_max_threads();
 
     // use per-token kernel, for each k,v token
     //  attn mask is a matrix of q_len(kv_len)
     buf_attn_w.resize<float>({B, H, q_len, kv_len});
+#if defined(HAVE_AVX2) && !defined(HAVE_AVX512F)
+    // avx2 will pre-compute the zero point and try to save the sub instruction in the dot_product,
+    //  but it seems not necessary for avx512. Possible reason may be that for avx2 the cost of dot_product
+    //  is larger than the memory access time, but for avx512 is not and the cost of pre-compute is a pure increase.
```

Comment:
Interesting tradeoff, but will the conclusion be different on following cases?
 - on AVX2 server with more cores
 - on AVX512 server, inside a VM with limited number of cores

Reply:
To cover these scenarios is difficult, if it can be changed at runtime it should be better, currently only use some test results as the heuristic.